### PR TITLE
Update the build system to use CMake instead of just Makefiles

### PR DESCRIPTION
**Pull request by jcanseco**

### CODE REVIEW

**Comment by jcanseco** [comment]
This PR is dependent on PR #23.

**Comment by joelagnel** [comment]
I see you did update the instructions, so that's good. I will further review this PR once you can update the PR where you do the restructuring and then refresh this one.

File: src/bpfd.h [hunk 1]
```diff
@@ -20,8 +20,8 @@
 #include <inttypes.h>
 
 #include "utils.h"
-#include "lib/bpf/libbpf.h"
 #include "base64.h"
+#include "bcc/libbpf.h"
```
Comment (joelagnel) [inline src/bpfd.h]: This change here is unrelated to the the commit (Using cmake). It should be made a part of the restructuring commit.

Comment (jcanseco) [reply src/bpfd.h]: This change was caused by a commit from PR #23 being included in this PR (since this PR is developed right on top of that PR). So in other words, this change _is_ part of the restructuring commit. Apologies for the confusion.

Comment (joelagnel) [reply src/bpfd.h]: No problem, yeah its hard to review dependent PRs on github and really confusing. Its more of a github issue. Anyway I think lets do one at a time and finish #23 first, then refresh these PRs and I'll review them. Thanks.

**Comment by jcanseco** [comment]
I went ahead and rebased this branch on top of 'restructure' from PR #23 so that it contains all the requested changes from that PR. It seemed like an easy change so I went ahead and did it.

**Comment by joelagnel** [comment]
Rebase on master again. Stale commits are again appearing in your PR list

File: INSTALL.md [hunk 2]
```diff
@@ -120,16 +120,22 @@ are welcome.
 Clone and build the BPFd sources:
 ```
 git clone git@github.com:joelagnel/bpfd.git; cd bpfd
+mkdir -p build; cd build
+cmake ..
 make
 ```
+
```
Comment (joelagnel) [inline INSTALL.md]: minor nit: Kill the new line?

File: toolchain-aarch64.cmake [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) Jazel Canseco, 2018
+# Licensed under the Apache License, Version 2.0 (the "License")
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR aarch64)
+
+set(CMAKE_C_COMPILER   /usr/bin/aarch64-linux-gnu-gcc-4.9)
```
Comment (joelagnel) [inline toolchain-aarch64.cmake]: This makes it bound to an absolute path. I don't like that. It should be able to pick up the compiler from $PATH

File: toolchain-aarch64.cmake [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+set(CMAKE_CXX_COMPILER /usr/bin/aarch64-linux-gnu-g++-4.9)
```
Comment (joelagnel) [inline toolchain-aarch64.cmake]: same here.

**Comment by jcanseco** [comment]
Made the requested changes. PTAL.

**Comment by joelagnel** [comment]
Why didn't you delete the old Makefile(s)? Please submit a follow up PR with that. I am merging this one.
